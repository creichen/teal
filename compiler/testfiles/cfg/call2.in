fun h() = {
  return 1;
}
fun g(y, z) = {
  return y;
}
fun f() = {
  return g(h(), h());
}

