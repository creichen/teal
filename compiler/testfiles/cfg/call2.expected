Entry#0 @ 1 :
  -> IntConstant#0
ReturnStmt#0 @ 2 :
  -> Exit#0
IntConstant#0 @ 2 :
  -> ReturnStmt#0
Exit#0 @ 3 :
Entry#1 @ 4 :
  -> Access#0
VarDecl#0 @ 4 :
VarDecl#1 @ 4 :
ReturnStmt#1 @ 5 :
  -> Exit#1
Access#0 @ 5 :
  -> ReturnStmt#1
Exit#1 @ 6 :
Entry#2 @ 7 :
  -> CallExpr#1
ReturnStmt#2 @ 8 :
  -> Exit#2
CallExpr#0 @ 8 :
  -> ReturnStmt#2
CallExpr#1 @ 8 :
  -> CallExpr#2
CallExpr#2 @ 8 :
  -> CallExpr#0
Exit#2 @ 9 :
